src/screens/DiscoveryScreenV2/elements/DiscoveryTab.js

```diff
@@ -1,44 +1,64 @@
 import * as React from 'react';
 import {Dimensions, Keyboard, Pressable, ScrollView, StyleSheet, Text, View} from 'react-native';
 
+import {useRoute} from '@react-navigation/core';
+import PropTypes from 'prop-types';
 import {colors} from '../../../utils/colors';
-import {fonts, normalizeFontSize} from '../../../utils/fonts';
+import {normalizeFontSize} from '../../../utils/fonts';
+import {setNavbarTitle} from '../../../context/actions/setMyProfileAction';
+import {Context} from '../../../context';
 
-const {width} = Dimensions.get('screen');
+const windowWidth = Dimensions.get('window').width;
+
+const DiscoveryTab = ({onChangeScreen, selectedScreen = 0, tabs}) => {
+  const [, dispatchNavbar] = React.useContext(Context).profile;
+  const route = useRoute();
 
-const DiscoveryTab = ({onChangeScreen, selectedScreen = 0}) => {
   const handleTabOnClicked = React.useCallback((index) => {
     Keyboard.dismiss();
     onChangeScreen(index);
   }, []);
 
-  const tabs = ['Users', 'Communities', 'Domains', 'News'];
+  const handleChangeTitle = (index = 0) => {
+    const title = ['Search Users', 'Your Communities', 'Your Domains'][index] || '';
+    setNavbarTitle(title, dispatchNavbar);
+  };
+
+  React.useEffect(() => {
+    if (route.name === 'Followings') handleChangeTitle();
+  }, []);
+
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `useEffect` hook at line 27 is missing dependencies. It's currently set to run only once when the component mounts, but it should also run whenever `route.name` changes. This could lead to unexpected behavior if `route.name` changes and the title doesn't update accordingly.
```diff
- React.useEffect(() => {
-   if (route.name === 'Followings') handleChangeTitle();
- }, []);
+ React.useEffect(() => {
+   if (route.name === 'Followings') handleChangeTitle();
+ }, [route.name]);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->

Reply:
useEffect only need to run once, and `route.name` will be never undefined, so it doesn't need to add dependecies there